### PR TITLE
Updated keycloak to 19.0.0

### PR DESCRIPTION
**Pull request by vidhya03**
fix https://github.com/jhipster/generator-jhipster/issues/19247

Upgrade keycloak to 19.0.0
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by DanielFran** [comment]
@vidhya03 Can you sign the CLA?

**Comment by vidhya03** [comment]
@vishal423  Could you please suggest how to run the tests locally. I am new to this. thanks a lot

File: generators/docker-compose/templates/realm-config/jhipster-realm.json.ejs [hunk 1]
```diff
@@ -2492,7 +2492,7 @@
     "clientOfflineSessionIdleTimeout": "0",
     "cibaInterval": "5"
   },
-  "keycloakVersion": "18.0.0",
+  "keycloakVersion": "19.0.0",
```
Comment (vishal423) [inline generators/docker-compose/templates/realm-config/jhipster-realm.json.ejs:2495]: I hope you diffed the new realm configurations to identify the version as the only change.

You also need to update the `generators/server/templates/src/main/docker/config/realm-config/jhipster-realm.json.ejs` with similar change

**Review by vishal423 — COMMENTED** [review]
I suggest to manually verify the following application flows:
- Monolithic application with Oauth2 - Login/Logout/Admin tab/entity tabs etc
- Micro-services application with Oauth2 (same flows as above) , JHipster Registry login
- Resource server flow (generate token and consume any secured api via token) to simulate the mobile clients

For failure tests, the best way is to copy the corresponding JDL/.yo-rc.json and regenerate locally to verify the end to end flow. However, if you test above flows, I think you may not need to look into these failures

File: generators/docker-compose/templates/realm-config/jhipster-realm.json.ejs (hunk 1 above)
Comment (vidhya03) [reply generators/docker-compose/templates/realm-config/jhipster-realm.json.ejs:2495]: Thank you so much Vishal,  please correct me if my understanding is not correct, actually 

```generators/server/templates/src/main/docker/config/realm-config/jhipster-realm.json.ejs```
 is resolved via  DOCKER_KEYCLOAK_VERSION constants

```/generators/generator-constants.js ```

image

Comment (vishal423) [reply generators/docker-compose/templates/realm-config/jhipster-realm.json.ejs:2495]: ok. In that case, you just need to check for diff to be included in the change

**Comment by mraible** [comment]
Is there anything I can do to help move this along?

**Comment by DanielFran** [comment]
Superseded by https://github.com/jhipster/generator-jhipster/pull/19540

**Pull request closed without merging**